base/task.jl

```diff
@@ -301,6 +341,17 @@ macro async(expr)
     end
 end
 
+# add a wait-able object to the sync pool
+macro sync_add(expr)
+    var = esc(sync_varname)
+    quote
+        local ref = $(esc(expr))
+        if $(Expr(:isdefined, var))
```

Comment:
this should perhaps not include the isdefined check and just assert that there is a sync pool?

Reply:
Ah, right.